tr1/carrier: add support for TR2-style item drops

lahm86 opened the pull request:
Resolves #1713.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This allows item drops to be defined in the level data in the same fashion as TR2, so any pickup item that is in the same position as an enemy will be carried by that enemy. The gameflow approach is retained for OG levels, and the provided gameflow setting will default to this.

For OG, Pierre in Tihocan and the enemies in Natla's Mines should continue to drop as normal. If the gameflow setting is enabled in OG, then they will not drop anything (and indeed the ape in room 82 of Tihocan will drop shotgun shells).

Attached is a custom level with every enemy type and some drops allocated to each. Similarly, if the gameflow setting is disabled here, then the items will appear normally (albeit some will float as they're positioned with the enemy).
[LEVEL1.zip](https://github.com/user-attachments/files/17383340/LEVEL1.zip)

lahm86 committed:
tr1/carrier: add support for TR2-style item drops

This allows item drops to be defined in the level data in the same
fashion as TR2, so any pickup item that is in the same position as an
enemy will be carried by that enemy. The gameflow approach is retained
for OG levels, and the provided gameflow setting will default to this.

Resolves #1713.

aredfan reviewed (approved):
I found an issue when tr2 item drops are enabled. The Qualopec mummy doesn't drop the pickup when Lara bumps into it.

Apart from that, LGTM. 👍

lahm86 committed:
tr1/carrier: test for Qualopec mummy on item drop

This moves the enemy status check when testing item drops into a
separate function for clarity, and checks additionally for the
Qualopec mummy, whose hitpoints will still be greater than 0 when Lara
touches it, and this still qualifies as a kill.

lahm86 commented:
> I found an issue when tr2 item drops are enabled. The Qualopec mummy doesn't drop the pickup when Lara bumps into it.
> 
> Apart from that, LGTM. 👍

Thank you, good catch. That should be fixed now.
